### ISSUE TITLE
[Combobox] Using keyboard, cannot navigate Chip with arrow keys initially

### ISSUE THREAD

**Issue by macandcheese** [Esri/calcite-design-system#6776]
### Actual Behavior

Similar to but slightly distinct from: https://github.com/Esri/calcite-components/issues/6750

After tabbing into the group of Chips - I cannot use the arrows to navigate the chip's close button. I need to advance to the end of the tab row using tab until I can begin to use the arrows.

### Expected Behavior

I would immediately be able to use Arrow Right to navigate through group of Chips

### Reproduction Sample

https://codepen.io/mac_and_cheese/pen/MWPKWXY?editors=1000

https://user-images.githubusercontent.com/4733155/231614920-8147a790-d92b-4013-bf9c-21fc124be639.mov

### Reproduction Steps

Open Codepen
- Use Tab to focus Combobox - first chip's close button will be focused first.
- Try to use Right Arrow to attempt to advance through Chip Close Buttons
- Notice that the focus doesn't change
- Use left arrow key, or tab again, and the right arrow will work.

### Reproduction Version

1.2

### Priority impact

p4 - not time sensitive

### Impact

@paulcpederson could you confirm if this is expected or a bug?

### Esri team

Calcite (design)

**Comment by geospatialem** [comment]
Related and could be approached together, possibly with `chip-group` as a refactor https://github.com/Esri/calcite-design-system/issues/6750 + https://github.com/Esri/calcite-design-system/issues/6777

**Comment by driskull** [comment]
Once you type something into the input, the arrow keys no longer function. You can't go back to edit a character without deleting one. Is this intentional? Seems like a usability/accessibility issue? cc @geospatialem 

1. tab or use arrow keys into the combobox
2. Keep tabbing or using arrow keys until you get to the input
3. type something
4. try to use arrow keys to edit what you typed. (failure)

Would this be a separate or related issue?

**Comment by driskull** [comment]
This is a component that could benefit from aria-activedescendant as well.

If we expect combobox to behave like a single component, then tabindex should only be active on a single element within it at any time. Arrow keys should be the primary form of navigation. I think this component needs some refactoring to handle that.

**Comment by paulcpederson** [comment]
> @paulcpederson could you confirm if this is expected or a bug?

Sorry I missed this, I think the behavior you discussed is a bug. I would expect to land on the input first when tabbing to the combobox with active tabs. Then arrows should always change the active element. That's just my 2¢, not sure if that is per the aria spec. 

Certainly when you have text entered into the input the arrow keys should either work as normal in an input or change the focus, in the pen they don't do either. 

> This is a component that could benefit from aria-activedescendant as well.

Yeah the combobox already uses active descendant, so likely would just be about extending that encompass the chips. Unless browsers have updated since this was built though, the challenge will be associating the active descendant id signifier across the shadow dom, that was impossible when this was built, but I haven't followed up in a while.

**Comment by driskull** [comment]
> Sorry I missed this, I think the behavior you discussed is a bug. I would expect to land on the input first when tabbing to the combobox with active tabs. Then arrows should always change the active element. That's just my 2¢, not sure if that is per the aria spec.

I think we need to discuss the expected behavior of the combobox. I think what Paul said here makes sense but that would mean taking away tabindex from the chips and using some sort of aria to let screen readers know which chip or dropdown element is selected.

Currently, calling `setFocus` on the combobox will always focus the text input. However, we may want to consider focusing the first focusable element which in some cases would be the first chip because it has a tabindex. So we should discuss removing these tabindexes so this isn't an issue because focusing the input by default makes the most sense. The  inconsistency here is because tabbing into the combobox produces different results than calling setFocus.

> Yeah the combobox already uses active descendant, so likely would just be about extending that to encompass the chips. Unless browsers have updated since this was built, the challenge will be associating the active descendant id signifier across the shadow dom. That was impossible when this was built, but I haven't followed up in a while.

Yeah that still may be an issue with the shadowDOM.

**Comment by driskull** [comment]
I think a roving tabindex might just be the solution here. Needs to have the chips tabindex set to -1 by default and make sure only one tabindex is 0 when navigating via keyboard.

**Comment by driskull** [comment]
There seems to be no way to keyboard to the clear button on the combobox either. :(

**Comment by DitwanP** [comment]
https://github.com/Esri/calcite-design-system/assets/60022782/5e7314d6-6a89-46f2-affb-59ce8de1604d

@driskull Once tabbed in I can immediately navigate with the arrow keys, but when I type I can't move between letters with arrow keys like you can elsewhere. If this is what we want then its good to go, just LMK.

**Comment by driskull** [comment]
@DitwanP that is how it currently is. This issue was created https://github.com/Esri/calcite-design-system/issues/9614
